Upgrade embedded GLEW to version 1.13.0

akien-mga opened the pull request:
Also drop all our customisations of the upstream files, by:
- Fixing the buggy Haiku setup (GLEW usage on Haiku should be investigated by users of this platform, grepping for `__HAIKU__` does give matches in the GLEW code, so it might be supported now)
- Use SCons magic to make a static build, instead of hardcoding it in the source files
- Also relies on SCons not to build `glew.c` when `GLEW_ENABLED` is not defined

Fixes #5087.

akien-mga committed:
Haiku: Do not define GLEW_ENABLED to undefine it later

Partial revert of f61eb5fd8e13642c82364f8ee66a0f6c791a4511.

akien-mga committed:
GLEW: Define static + enabled and includes via SCons

This allows us not to have to hack our definitions in the upstream files,
making it easier to upgrade to newer versions in the future.

For the include paths to work, the headers are moved to a GL subfolder to
match their upstream location.

vnen commented:
Fails to compile on MSVC: `.\drivers/gles2/rasterizer_gles2.h(51): fatal error C1083: Cannot open include file: 'gl_context/glew.h': No such file or directory`

akien-mga commented:
Thanks, I've fixed that locally, I'll soon push a rebased version.

I also noticed that I had forgotten to readd the `#ifdef GLEW_ENABLED` in `glew.c`, so I'm working on a SCons-based variant to have the same behaviour without having to touch the upstream source file.

akien-mga committed:
GLEW: Update to upstream version 1.13.0 (pristine)

This removes a custom redefinition of glewGetProcAddress,
which was apparently necessary for older Macs with buggy OpenGL
(see #5087).

The added source files are as provided in the upstream tarball.

akien-mga commented:
Rebased and added a commit for the `GLEW_ENABLED` stuff :)

akien-mga commented on drivers/gl_context/SCsub:
```diff
@@ -2,7 +2,8 @@ Import('env')
-
-env_drivers.Append(CFLAGS = ['-DGLEW_STATIC'])
-env_drivers.Append(CPPPATH = ['.'])
+if (env['glew'] == 'yes'):
```
Apparently that won't work if `env['glew']` was not defined, so I'll have to use `if ('glew' in env and env['glew'] == 'yes'`.

J08nY replied:
Isn't SConscript essentially python? In that case, if `env` is a python `dict`you can use `env.get('glew')` which returns `None` if it's not there and the value if it is.

akien-mga replied:
Thanks, I'll use that.
